### PR TITLE
fix: gate release based on releasability check.

### PR DESCRIPTION
**Pull request by peternhale**

### CODE REVIEW

File: src/commands/check-if-releasable.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# How to author commands: https://circleci.com/docs/2.0/reusing-config/#authoring-reusable-commands
+description: >
+  Determine if a release is necessary. Runs sfdx npm:release:validate, which returns a true if a release is needed.
+  Command will set env var NEEDS_RELEASE accordingly
+
+steps:
+  - run:
+      name: Release validate
+      command: |
+        NEEDS_RELEASE=$(sfdx npm:release:validate)
```
Comment (peternhale) [inline src/commands/check-if-releasable.yml:15]: Run release:validate for the current repo

File: src/commands/check-if-releasable.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        echo "Release Needed:${NEEDS_RELEASE}"
+        echo 'export NEEDS_RELEASE="${NEEDS_RELEASE}"' >> $BASH_ENV
```
Comment (peternhale) [inline src/commands/check-if-releasable.yml:17]: Save results in to current env

File: src/jobs/release-lerna-packages.yml [hunk 3]
```diff
@@ -77,10 +77,18 @@ steps:
         - ~/.cache/yarn
         - ~/.config/yarn
 
+  # - run npm:release:validate to see if a release is necessary - will set env var NEEDS_RELEASE
```
Comment (peternhale) [inline src/jobs/release-lerna-packages.yml:80]: run the step

File: src/jobs/release-lerna-packages.yml [hunk 4]
```diff
@@ -97,6 +105,7 @@ steps:
   - when:
       condition:
         and:
+          - equal: [true, $NEEDS_RELEASE]
```
Comment (peternhale) [inline src/jobs/release-lerna-packages.yml:108]: gate each step involved in a release

File: src/jobs/release-package.yml [hunk 5]
```diff
@@ -100,10 +100,18 @@ steps:
         - ~/.cache/yarn
         - ~/.config/yarn
 
+  # - run npm:release:validate to see if a release is necessary - will set env var NEEDS_RELEASE
```
Comment (peternhale) [inline src/jobs/release-package.yml:103]: same here

**Review by mdonnalley — APPROVED** [review]

File: src/jobs/release-lerna-packages.yml [hunk 6]
```diff
@@ -77,10 +77,18 @@ steps:
         - ~/.cache/yarn
         - ~/.config/yarn
 
+  # - run npm:release:validate to see if a release is necessary - will set env var NEEDS_RELEASE
+  - when:
+      condition:
+        not: <<parameters.dryrun>>
+      steps:
+        check-if-releasable
   # - if environment looks like it is prepped to do change cases, and we're on "latest" tag
```
Comment (mshanemc) [inline src/jobs/release-lerna-packages.yml:86]: does it actually check CTC envs?

Comment (peternhale) [reply src/jobs/release-lerna-packages.yml:86]: @mshanemc If "it" is `npm:release:validate` then it does the same checks as the release management release cmd, but without the npm tag/publish, etc. The check just captures if the current project should be released or not and persists that determination in the env var "NEEDS_RELEASE".

**Review by mshanemc — APPROVED** [review]
approved, 1 non-blocking question

**Pull request merged**